docs: Add Tags documentation for Contact management (#647)
mautic/user-documentation: docs/contacts/tags.rst

> .. vale off

Author: **Citation:** Referenced `Modify Contact's tags` action and `Contact Tags` condition from the Campaign Builder documentation to ensure consistency in describing Tags usage within Campaigns.
[View source](docs/campaigns/campaign_builder.rst)

> .. vale off

Author: **Citation:** Referenced the `Modify Contact's Tags` Form action documentation to ensure consistency in describing Tags usage within Forms.
[View source](docs/components/forms.rst)

Tags
####

.. vale on

Tags are flexible labels you can add to or remove from Contacts at any time. Unlike Segments, which group Contacts based on filters, Tags let you manually categorize Contacts for quick identification, filtering, and triggering Campaign actions.

Managing Tags

> *************

Reviewer: Let's wrap this in `vale off` and `vale on`.

```suggestion
.. vale off

Managing Tags
*************

.. vale on
```

Creating Tags

> =============

Reviewer: ```suggestion
.. vale off

Creating Tags
=============

.. vale on
```

Tags don't require pre-creation. You can create new Tags directly when applying them to Contacts - type a new Tag name in any Tag field, and Mautic creates it automatically.

To view existing Tags, navigate to a Contact's record and look at the Tags field, or use the Tag filter when searching Contacts.

Adding Tags to Contacts

> =======================

Reviewer: ```suggestion
.. vale off

Adding Tags to Contacts
=======================

.. vale on
```

Manual assignment
-----------------

1. Navigate to the Contact's record.
2. Click **Edit** to open the Contact edit form.
3. In the **Tags** field, start typing to search existing Tags or enter a new Tag name.
4. Press Enter or select from the dropdown to apply the Tag.

> 5. Click **Save** to save the changes.

Reviewer: The word "form" here is general, not Mautic's feature. So, let's wrap this in vale off/on.

```suggestion
.. vale off

#. Navigate to the Contact's record.
#. Click **Edit** to open the Contact edit form.
#. In the **Tags** field, start typing to search existing Tags or enter a new Tag name.
#. Press Enter or select from the dropdown to apply the Tag.
#. Click **Save** to save the changes.

.. vale on
```

Batch updates
-------------

1. Navigate to the **Contacts** section.
2. Use filters to find the Contacts you want to update.
3. Select the checkboxes next to the desired Contacts.
4. Click the green arrow at the top of the column.
5. Select **Change Tags** from the dropdown menu.
6. Choose Tags to add or remove.

> 7. Click **Save**.

Reviewer: ```suggestion
#. Navigate to the **Contacts** section.
#. Use filters to find the Contacts you want to update.
#. Select the checkboxes next to the desired Contacts.
#. Click the green arrow at the top of the column.
#. Select **Change Tags** from the dropdown menu.
#. Choose Tags to add or remove.
#. Click **Save**.
```

Using the tracking script
-------------------------

You can also add or remove Tags from Contacts using the tracking script or tracking pixel. For more information, see :ref:`Tags <contacts/manage_contacts:Tags>` in the Contact Monitoring documentation.

Using Tags in Campaigns

> ***********************

Reviewer: ```suggestion
.. vale off

Using Tags in Campaigns
***********************

.. vale on
```

.. vale off

Modify Contact's Tags action
============================

.. vale on

Use the **Modify Contact's Tags** action to add or remove Tags from Contacts as they progress through a Campaign.

1. In the Campaign Builder, click the connector below an event.
2. Select **Action**.
3. Choose **Modify Contact's Tags**.
4. Select Tags to add or remove.
5. Click **Add** to save the action.

> This action is useful for marking Contacts who have reached certain stages, completed specific actions, or require follow-up.

Reviewer: ```suggestion
This action is useful for marking Contacts who have reached certain Stages, completed specific Actions, or require follow-up.
```

Contact Tags condition

> ======================

Reviewer: ```suggestion
.. vale off

Contact Tags condition
======================

.. vale on
```

Use the **Contact Tags** condition to create different Campaign paths based on whether a Contact has specific Tags.

1. In the Campaign Builder, click the connector below an event.
2. Select **Condition**.
3. Choose **Contact Tags**.

> 4. Configure which Tags to check for.

Reviewer: ```suggestion
4. Configure which Tags to validate.
```

> 5. Click **Add** to save the condition.

Reviewer: We're using `#.` for ordered list.

```suggestion
#. In the Campaign Builder, click the connector below an event.
#. Select **Condition**.
#. Choose **Contact Tags**.
#. Configure which Tags to check for.
#. Click **Add** to save the condition.
```

Contacts with matching Tags follow the **Yes** path, while others follow the **No** path.

Using Tags in Forms

> *******************

Reviewer: ```suggestion
.. vale off

Using Tags in Forms
*******************

.. vale on
```

You can automatically add or remove Tags when a Contact submits a Form.

1. Navigate to **Components > Forms** and edit a Form.
2. In the Form builder, go to the **Actions** tab.
3. Click **Add new action** and select **Modify Contact's Tags**.
4. Select Tags to add or remove upon Form submission.

> 5. Click **Add** to save the action.

Reviewer: ```suggestion
#. Navigate to **Components > Forms** and edit a Form.
#. In the Form builder, go to the **Actions** tab.
#. Click **Add new action** and select **Modify Contact's Tags**.
#. Select Tags to add or remove upon Form submission.
#. Click **Add** to save the action.
```

This is helpful for tagging Contacts based on which Forms they complete, such as marking someone as interested in a specific product.

Using Tags in Segments
**********************

You can use Tags as filters when building Segments:

1. Navigate to **Segments** and create or edit a Segment.
2. Go to the **Filters** tab.
3. Add a filter and select **Tags** from the Contact field options.
4. Choose the operator - for example, **Includes**, **Excludes**, **Empty**, or **Not empty**.
5. Select the Tag or Tags to filter by.

> 6. Click **Save** to apply the filter.

Reviewer: ```suggestion
#. Navigate to **Segments** and create or edit a Segment.
#. Go to the **Filters** tab.
#. Add a filter and select **Tags** from the Contact field options.
#. Choose the operator - for example, **Includes**, **Excludes**, **Empty**, or **Not empty**.
#. Select the Tag or Tags to filter by.
#. Click **Save** to apply the filter.
```

This creates Segments of Contacts who have - or don't have - specific Tags applied.

Searching Contacts by Tags

> **************************

Reviewer: ```suggestion
.. vale off

Searching Contacts by Tags
**************************

.. vale on
```

You can search for Contacts with specific Tags using the search bar:

- To find Contacts with a specific Tag, use: ``tag:tagname``

> - To find Contacts without a specific Tag, use: ``!tag:tagname``

Reviewer: We use `*` for unordered list.

```suggestion
* To find Contacts with a specific Tag, use ``tag:tagname``
* To find Contacts without a specific Tag, use ``!tag:tagname``
```

> Replace ``tagname`` with the actual name of the Tag. If the Tag name contains spaces, wrap it in quotes: ``tag:"tag name"``

Reviewer: Let's stay away from `:` when possible.

```suggestion
Replace ``tagname`` with the actual name of the Tag. If the Tag name contains spaces, wrap it in quotes. For example, ``tag:"tag name"``.
```

Best practices
**************

- **Use consistent naming conventions** - Establish naming rules to avoid duplicates like "Webinar" and "webinar-attendee" for similar purposes.
- **Document your Tag taxonomy** - Keep a record of what each Tag means and when to use it.
- **Avoid over-tagging** - Too many Tags can become difficult to manage. Consider whether a Segment or Custom Field might be more appropriate.
- **Review Tags regularly** - Periodically audit your Tags to remove outdated or unused ones.

> - **Combine with Segments** - Use Tags for quick manual labeling and Segments for dynamic, filter-based grouping.

Reviewer: Use single quote for word(s) and double quotes for quotes.

```suggestion
* **Use consistent naming conventions** - Establish naming rules to avoid duplicates, such as 'Webinar' and 'webinar-attendee', for similar purposes.
* **Document your Tag taxonomy** - Keep a record of what each Tag means and when to use it.
* **Avoid over-tagging** - Too many Tags can become difficult to manage. Consider whether a Segment or Custom Field might be more appropriate.
* **Review Tags regularly** - Periodically audit your Tags to remove outdated or unused ones.
* **Combine with Segments** - Use Tags for quick manual labeling and Segments for dynamic, filter-based grouping.
```

Managing Tags
*************

.. vale on

.. vale off

Creating Tags
=============

.. vale on

Tags don't require pre-creation. You can create new Tags directly when applying them to Contacts - type a new Tag name in any Tag field, and Mautic creates it automatically.

To view existing Tags, navigate to a Contact's record and look at the Tags field, or use the Tag filter when searching Contacts.

.. vale off

Adding Tags to Contacts
=======================

.. vale on

Manual assignment
-----------------

.. vale off

#. Navigate to the Contact's record.
#. Click **Edit** to open the Contact edit form.
#. In the **Tags** field, start typing to search existing Tags or enter a new Tag name.
#. Press Enter or select from the dropdown to apply the Tag.
#. Click **Save** to save the changes.

.. vale on

Batch updates
-------------

#. Navigate to the **Contacts** section.
#. Use filters to find the Contacts you want to update.
#. Select the checkboxes next to the desired Contacts.
#. Click the green arrow at the top of the column.
#. Select **Change Tags** from the dropdown menu.
#. Choose Tags to add or remove.
#. Click **Save**.

Using the tracking script
-------------------------

> You can also add or remove Tags from Contacts using the tracking script or tracking pixel. For more information, see :ref:`Tags <contacts/manage_contacts:Tags>` in the Contact Monitoring documentation.

Reviewer: Do we have Contact Monitoring docs?

> You can also add or remove Tags from Contacts using the tracking script or tracking pixel. For more information, see :ref:`Tags <contacts/manage_contacts:Tags>` in the Contact Monitoring documentation.

Reviewer: @promptless-for-oss we can directly link the "Contact tracking" section with custom target here.

- In `contacts/manage_contacts.rst`, add `.. _Contact tracking:` right before "Contact tracking" header.
- Then fix the link as follow:

```suggestion
You can also add or remove Tags from Contacts using the tracking script or tracking pixel. For more information, see :ref:`Contact tracking` section.
```

.. vale off

Using Tags in Campaigns
***********************

.. vale on

.. vale off

Modify Contact's Tags action
============================

.. vale on

Use the **Modify Contact's Tags** action to add or remove Tags from Contacts as they progress through a Campaign.

1. In the Campaign Builder, click the connector below an event.
2. Select **Action**.
3. Choose **Modify Contact's Tags**.
4. Select Tags to add or remove.
5. Click **Add** to save the action.

This action is useful for marking Contacts who have reached certain Stages, completed specific Actions, or require follow-up.

.. vale off

Contact Tags condition
======================

.. vale on

Use the **Contact Tags** condition to create different Campaign paths based on whether a Contact has specific Tags.

#. In the Campaign Builder, click the connector below an event.
#. Select **Condition**.
#. Choose **Contact Tags**.

> #. Configure which Tags to check for.

Reviewer: Use another word than "check" to address Vale warning.

#. Click **Add** to save the condition.

Contacts with matching Tags follow the **Yes** path, while others follow the **No** path.

.. vale off

Using Tags in Forms
*******************

.. vale on

You can automatically add or remove Tags when a Contact submits a Form.

#. Navigate to **Components > Forms** and edit a Form.
#. In the Form builder, go to the **Actions** tab.
#. Click **Add new action** and select **Modify Contact's Tags**.
#. Select Tags to add or remove upon Form submission.
#. Click **Add** to save the action.

This is helpful for tagging Contacts based on which Forms they complete, such as marking someone as interested in a specific product.

> Using Tags in Segments
> **********************

Reviewer: ```suggestion
.. vale off

Using Tags in Segments
**********************

.. vale on
```
